### PR TITLE
Avoid attaching security groups to NLB in unsupported regions and fix integration tests

### PR DESCRIPTION
**Pull request by hanwen-pcluste**
See commits descriptions for details

### Description of changes
* Tests in the unsupported regions passed

### Tests
* Describe the automated and/or manual tests executed to validate the patch.
* Describe the added/modified tests.

### References
* Link to impacted open issues.
* Link to related PRs in other packages (i.e. cookbook, node).
* Link to documentation useful to understand the changes.

### Checklist
- Make sure you are pointing to **the right branch**.
- If you're creating a patch for a branch other than `develop` add the branch name as prefix in the PR title (e\.g\. `[release-3.6]`).
- Check all commits' messages are clear, describing what and why vs how.
- Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by gmarciani — CHANGES_REQUESTED** [review]
See comments in https://github.com/aws/aws-parallelcluster/pull/6447#pullrequestreview-2343477585

File: cloudformation/ad/ad-integration.yaml [hunk 1]
```diff
@@ -708,9 +708,6 @@ Resources:
                   - iam:UploadServerCertificate
                   - iam:TagServerCertificate
                 Resource: !Sub arn:${AWS::Partition}:iam::${AWS::AccountId}:server-certificate/*
-                Condition:
```
Comment (gmarciani) [inline cloudformation/ad/ad-integration.yaml:711]: Why are we removing this condition?

**Review by hgreebe — APPROVED** [review]

File: cloudformation/ad/ad-integration.yaml (hunk 1 above)
Comment (hanwen-pcluste) [reply cloudformation/ad/ad-integration.yaml:711]: In commit descriptions: iso regions don't support tagging certificates. Therefore, the condition in IAM policy is too restrictive in iso reginos

Comment (gmarciani) [reply cloudformation/ad/ad-integration.yaml:711]: Then the restriction should be removed only if in ISO regions. Let's do this in a follow up PR

**Review by gmarciani — APPROVED** [review]

**Pull request merged**